src/wwwroot/controllers/TemplateCtrl.js

```diff
@@ -39,39 +43,56 @@
 
     function load(templateId) {
       $scope.loadInProgress = true;
-
-      return templates.getTemplateWithBody(templateId).then(function (result) {
-        $scope.template.fromName = result.from_name;
-        $scope.template.fromEmail = result.from_email;
-        $scope.template.subject = result.subject;
-        $scope.template.name = result.name;
-        $scope.template.id = result.id;
-        $scope.template.content = result.body;
+      return templates.getTemplate(templateId).then(function (template) {
+        $scope.template.fromName = template.from_name;
+        $scope.template.fromEmail = template.from_email;
+        $scope.template.subject = template.subject;
+        $scope.template.name = template.name;
+        $scope.template.id = template.id;
+        $scope.isHtmlTemplate = template.bodyType == "rawHtml" || (template.bodyType == "empty" && !$rootScope.forceMsEditor);
+        if (!$scope.isHtmlTemplate) {
+          $scope.template.content = null;
+        } else {
+          return templates.getTemplateBody(templateId).then(function(templateBody) {
+            $scope.template.content = templateBody.html;
+          });
+        }
       }).finally(function () {
         $scope.loadInProgress = false;
       });
     }
 
     function save() {
+      if ($scope.form.$invalid) {
+        return;
+      }
       $scope.saveInProgress = true;
-      return templates.save({
+      var templateModel = {
         from_name: $scope.template.fromName,
         from_email: $scope.template.fromEmail,
         subject: $scope.template.subject,
         body: $scope.template.content,
         name: $scope.template.name,
         id: $scope.template.id
-      }).finally(function () {
-        $scope.saveInProgress = false;
-      });
-    }
-
-    function saveHtmlRaw() {
-      if ($scope.form.$invalid) {
-        return;
       }
-      return save().then(function (result) {
+      var isCreating = !$scope.template.id;
+      var saveTemplatePromise = isCreating ? templates.createTemplate(templateModel) : templates.editTemplate(templateModel);
+      return saveTemplatePromise.then(function(templateId) {
+        if ($scope.isHtmlTemplate) {
+          return templates.editTemplateBody(templateId, templateModel.name, templateModel.body).then(function() {
+            return templateId;
+          })
+          .catch(function (reason) {
+            // TODO: Show a special message error when template was update but body fails
+            return $q.reject(reason);
+          });
+        } else {
+          return templateId;
+        }
+        
+      }).then(function (result) {
         $location.path('/templates');
```

Comment:
It is being executed even when we save to continue editing with MSEditor, please apply a little refactoring here to fix that.

Reply:
👍